src/setupWorker/start/createStart.ts

```diff
@@ -61,6 +64,12 @@ export const createStart = (context: SetupWorkerInternalContext) => {
       const [worker, registration] = instance
 
       if (!worker) {
+        if (options?.serviceWorkerMatcher) {
+          console.warn(
+            `[MSW] worker.start() was provided a serviceWorkerMatcher predicate that failed to return a worker instance. Please verify your configuration.`,
```

Comment:
This warning message is misleading: the predicate doesn't return a worker instance, but finds it in the list of currently active service workers.

Reply:
Maybe let's drop this warning for now? I'm not sure what value it brings, as when the code fails to locate the worker you will get a registration error anyway, wouldn't you?

Reply:
In this case, if someone only supplied a `findWorker` that failed to find a worker, it'd return `null` and there would be no error because the worker technically registers. 

Commented relevant code: 

```js
const [error, instance] = await until<ServiceWorkerInstanceTuple>(
    async () => {
      const registration = await navigator.serviceWorker.register(url, options) // 1. registers and won't throw
      return [
        // Compare existing worker registration by its worker URL,
        // to prevent irrelevant workers to resolve here (such as Codesandbox worker).
        getWorkerByRegistration(registration, absoluteWorkerUrl, findWorker), // 2. will be null with a bad findWorker option
        registration,
      ]
    },
  )
```

**Other options:**
We could throw an error if we fail to find a worker here after registration instead of handling it like I am proposing, but I don't think I like that idea because the worker would technically be registered already, and we'd then want to consider unregistering it? From a DX perspective, I'd prefer to keep the mockServiceWorker on the window if possible so it reactivates faster when the `findWorker` method is fixed.

Let me know what you think

Reply:
I think the `if` predicate or the message is wrong. 
`options?.findWorker` doesn't execute the predicate method, it just verifies that the predicate is provided.

Reply:
Well, in this case we're just checking to see if it was provided, so that is the intended behavior. `getWorkerInstance` calls `findWorker`. This does actually bring up another unhandled case (ben's original one).

Maybe stepping away from the code is a good idea. Here are the currently possible scenarios:

1. Worker is registered and then found
    - Mocking enabled 🎉 
2. Worker is provided a nonexistent file
    - `throw`s error, is caught and logged with DX message
3. Worker is registered, but then not found due to strict `scriptURL === mockServiceWorkerUrl`. 
    - unhandled, console would be empty
4. Worker is registered, but not found when provided `findWorker`
    - `console.warn`s to check configuration

I'm very much open to suggestions on how to improve this and appreciate the input. I can't currently think of any better language to improve the message(s).

Reply:
Per 👆 , this could be an idea:

```js
if (!worker) {
        if (options?.findWorker) {
          console.warn(
            `[MSW] worker.start() registered the service worker, but was provided a findWorker predicate that failed to find a worker instance. Please verify your configuration.`,
          )
        } else {
          console.warn(`\
[MSW] worker.start() registered the service worker, but was unable to find a worker instance. 

You can most likely resolve this by using calling worker.start() with the findWorker option. 

[link to docs].\
          `)
        }

        return null
      }
```

Reply:
Thanks for the clarification 🙏

Reply:
@msutkowski I liked that different message based on the `findWorker` option. Added that in the latest commit. Also, converted the warning to an error, because in either of those cases the operability of the library is compromised.